### PR TITLE
Add oneplus2

### PR DESCRIPTION
**Pull request by Vince1171**
Add UT and LineageOS support for the OnePlus 2

### CODE REVIEW

File: v1/oneplus2.json
Comment (NeoTheThird) [inline v1/oneplus2.json:9]: That just uses `fastboot oem unlock`, the installer should be able to handle that already.

**Review by NeoTheThird — CHANGES_REQUESTED** [review]
Good job, thanks! Consider changing the oem lock thing. Also, this should not be merged before 0.7.x-beta hits the shelves.

File: v1/oneplus2.json
Comment (Vince1171) [reply v1/oneplus2.json:9]: I haven't found example for the unlocking part
is it something like that?
```
"steps": [
      {
          "type": "fastboot:oem_unlock"
      }
]
```

Comment (Vince1171) [reply v1/oneplus2.json:9]: I see there is some kind of lock detection (src/main.js line 342) but I don't know how to trigger/manage it :/

Comment (NeoTheThird) [reply v1/oneplus2.json:9]: Lock your device (if it supports it `sudo oem lock`) and run the installer again. The `fastboot:flash` step will then run into an error that [will be recognized](https://github.com/ubports/promise-android-tools/blob/master/src/fastboot.js#L48) by our library. That'll cause the installer to jump out of the install step chain to prompt the user for permission to unlock. If the user clicks ok, the installer will run `fastboot oem unlock` and resume the install at the last fastboot flash step. Usually very reliable. There are many different errors devices can throw (because there's no standardization in fastboot), but if i get a report for a new unhandled error one, i know how to proceed and can fix it in the next installer release.

As a way forward, i propose just removing the instruction. It should usually just work.

Comment (Flohack74) [reply v1/oneplus2.json:9]: Take care, locking and unlocking the bootloader might wipe your data ;)

Comment (Vince1171) [reply v1/oneplus2.json:9]: @NeoTheThird there is an issue with "lock detection"
the error returned by the bootloader is different from the one in [fastboot.js](https://github.com/ubports/promise-android-tools/blob/master/src/fastboot.js#L48)
```
Error: fastboot:flash: Error: Flashing failed: {"error":{"code":1},"stderr":"FAILED (remote: 'device is locked. Cannot flash images')fastboot: error: Command failed"}
```
NOTE: Settings > Developer Options > OEM unlocking is enabled

@Flohack74 Yes, unlocking the bootloader delete /data... I'll add a warning about that in thee oneplus2 config

Comment (NeoTheThird) [reply v1/oneplus2.json:9]: @Vince1171 no, don't add a warning, that's default behavior. I'll add handling for that error. If you want, you can leave the oem unlock note in for now, but we can remove it again soon.

Comment (Vince1171) [reply v1/oneplus2.json]: @NeoTheThird  I've changed the message
let's wait for the merge of ubports/promise-android-tools#52 before merging this one

Comment (NeoTheThird) [reply v1/oneplus2.json]: >  I've changed the message

ok. i'd have removed it completely, as there's generic instructions for this, but it's ok for me if you want to keep it like this.

>  let's wait for the merge of ubports/promise-android-tools#52 before merging this one

As you wish. I'll try to squeeze in another 0.7 installer release with that included before entering QA for 0.8, so you don't have to wait a week.

Comment (NeoTheThird) [reply v1/oneplus2.json]: There you go, a prerelease: https://github.com/ubports/ubports-installer/releases/tag/0.7.4-beta

**Review by NeoTheThird — COMMENTED** [review]
Can be merged as soon as [0.7.4-beta](https://github.com/ubports/ubports-installer/releases/tag/0.7.4-beta) leaves QA.

File: v1/oneplus2.json
Comment (Vince1171) [reply v1/oneplus2.json]: done already, I was quicker :P
It works perfectly thank you :D

Comment (Vince1171) [reply v1/oneplus2.json]: and I've remove the oem unlock message

**Comment by Vince1171** [comment]
> Can be merged as soon as [0.7.4-beta](https://github.com/ubports/ubports-installer/releases/tag/0.7.4-beta) leaves QA.

thanks
The Installer is just awesome btw :D

**Comment by NeoTheThird** [comment]
>  The Installer is just awesome btw :D
thanks :) love your work on these new ports 🤝

**Review by NeoTheThird — APPROVED** [review]
Alright, 0.7.4-beta has been released. All clear from your side @Vince1171?

**Comment by Vince1171** [comment]
> Alright, 0.7.4-beta has been released. All clear from your side @Vince1171?

All clear and good to go Captain

**Pull request merged**
